updated workflow/main.yml

RasmusBC59 opened the pull request:
updated workflow/main.yml

RasmusBC59 committed:
Telemetry False

RasmusBC59 committed:
added function myadd

RasmusBC59 committed:
added test fot myadd

RasmusBC59 committed:
Update hello_world.py

added funct description

RasmusBC59 committed:
Create mainbla.yml

bla

RasmusBC59 committed:
Update main.yml

fixed indention

RasmusBC59 committed:
Update main.yml

changed path to environment.yml

RasmusBC59 committed:
Update main.yml

conda init bash

RasmusBC59 committed:
Update main.yml

source activate

RasmusBC59 committed:
Update main.yml

more echo

RasmusBC59 committed:
Update main.yml

echo "$(conda env list)"

RasmusBC59 committed:
Update main.yml

deactivate base

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

source to conda

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

split in two

RasmusBC59 committed:
Update main.yml

echo path

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

internet fix

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

setup-miniconda@v1.7.0

RasmusBC59 committed:
Update main.yml

conda info

RasmusBC59 committed:
Update environment.yml

added: pytest, pytest-cov, mypy

RasmusBC59 committed:
Update main.yml

mypy

RasmusBC59 committed:
Update main.yml

pytest

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update environment.yml

removed test re

RasmusBC59 committed:
Update main.yml

pytest

RasmusBC59 committed:
Update main.yml

Build The Docs

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

comment out fail

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

make nice

RasmusBC59 committed:
Merge pull request #4 from RasmusBC59/test

Test

RasmusBC59 committed:
Update main.yml

if: ${{succeededOrFailed()}}

RasmusBC59 committed:
Update main.yml

export SPHINXOPTS

RasmusBC59 committed:
Update main.yml

removed SuccededOrFailed

RasmusBC59 committed:
Update main.yml

RasmusBC59 committed:
Update main.yml

make html

jenshnielsen commented on .github/workflows/main.yml:
```diff
@@ -22,34 +22,48 @@ jobs:
+          conda update -n base -c defaults conda
+          conda info
+          conda list
+      - name: pip_install_test_requirements 
```
Is there any reason for installing conda? It seems like the package is installed with pip anyway and conda is unused

RasmusBC59 replied:
conda is installed to setup the environment pulsequantum. 
There is not much in the environment yet, but there will be as the packed evolves. 
Or have I misunderstood?

jenshnielsen replied:
There are two alternative package mangers for python conda and pip. You are first installing conda and then using pip to install the package and the dependencies. That looks a bit odd. I would start with a pure pip powered environment and if I use conda only use conda as much as possible without using pip. 
The plottr envs https://github.com/toolsforexperiments/plottr/blob/master/.github/workflows/python-app.yml and here https://github.com/toolsforexperiments/plottr/blob/master/.github/actions/install-dependencies-and-plottr/action.yml only used pip

RasmusBC59 replied:
I see you point :-) I was taking guidance from a azure pipline templat (see below) I will try to mimic the plottr workflow instead . 
```` python
     steps:
    - powershell: | # for conda executable and binaries like sqlite (which pytest-cov uses)
        Write-Host "##vso[task.prependpath]$env:CONDA\Library\mingw-w64\bin"
        Write-Host "##vso[task.prependpath]$env:CONDA\Library\usr\bin"
        Write-Host "##vso[task.prependpath]$env:CONDA\Library\bin"
        Write-Host "##vso[task.prependpath]$env:CONDA\Scripts"
        Write-Host "##vso[task.prependpath]$env:CONDA\bin"
      displayName: Add conda to PATH
    - bash: |
        conda update -n base conda pip setuptools -y &&
        conda init --all
      displayName: Conda setup
    - bash: |
        sed -i "s/\(enabled *= *\).*/\1False/" ./pulsequantum/conf/telemetry.ini
      name: disable_telemetry
      displayName: Disable telemetry
    - bash: |
        conda --version &&
        conda env create --file environment.yml &&
        source activate pulsequantum &&
        pip install -r test_requirements.txt &&
        pip install .
      name: install_package
      displayName: Install Package
````

jenshnielsen replied:
The idea for qcodes was to test with conda on azure pipelines and pip on travis.
However there is only one version of the test and docs requirements since those are not really used by end users

RasmusBC59 committed:
Update main.yml

removed #

RasmusBC59 committed:
Pipins (#5)

* added new workflow and action

* added requirements

* path to pytest and mypy

* fix run mypy

* fix yml syntax

* added pytest

* Update python-app.yml

changed  pytest  path

* Update python-app.yml

echo pwd

* Update python-app.yml

echo some more

* Update python-app.yml

echo ls

* Update python-app.yml

test ->tests

* Update python-app.yml

* telemetry false

* moved conftets

* Delete conftest.py

deleted conftest

* Update python-app.yml

try new pytest line

* buld docomentation

* added _static

* added opencensus-ext-azure to requirments

* install pulsequantum

* do not test py36

* pytest and htmlfast

* echo pip list

* action pulsequantum

* Update python-app.yml

* conda install pandoc

* only py37

* echo pandoc version

* echo pandoc version

* Update python-app.yml

* Update python-app.yml

pandoc action test 1

* cleaning of unused lines

* htmlfast -> html

* Delete action.yml

* dont conda install pandoc

* removed pandoc from requirements

RasmusBC59 committed:
python-app.yml run on main
